packages/fx-core/src/component/driver/file/createOrUpdateJsonFile.ts

```diff
@@ -59,27 +58,22 @@ export class CreateOrUpdateJsonFileDriver implements StepDriver {
   }
 
   private async handler(
-    args: GenerateAppsettingsArgs,
+    args: GenerateJsonArgs,
     context: DriverContext
   ): Promise<{
     output: Map<string, string>;
     summaries: string[];
   }> {
     try {
       this.validateArgs(args);
-      const appsettingsPath = getAbsolutePath(args.target, context.projectPath);
-      if (!(await fs.pathExists(appsettingsPath))) {
-        // try to copy appsettings.json
-        const appsettingsTemplatePath = getAbsolutePath("appsettings.json", context.projectPath);
-        if (!fs.existsSync(appsettingsTemplatePath)) {
-          await fs.writeFile(appsettingsPath, "{}");
-        } else {
-          await fs.copyFile(appsettingsTemplatePath, appsettingsPath);
-        }
+      const jsonFilePath = getAbsolutePath(args.target, context.projectPath);
+      if (!(await fs.pathExists(jsonFilePath))) {
+        await fs.ensureFile(jsonFilePath);
+        await fs.writeFile(jsonFilePath, "{}", "utf-8");
       }
-      const appSettingsJson = JSON.parse(fs.readFileSync(appsettingsPath, "utf-8"));
-      this.replaceProjectAppsettings(appSettingsJson, args.appsettings);
-      await fs.writeFile(appsettingsPath, JSON.stringify(appSettingsJson, null, "\t"), "utf-8");
+      const jsonContent = commentJson.parse(fs.readFileSync(jsonFilePath, "utf-8").toString());
```

Comment:
Perhaps it can be changed to an async read, given this is an async function.

Reply:
Thanks for the advice. Done.